src/seed_tools/utils/study_json_utils.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import { promises as fs } from 'fs';
+import Result from 'src/base/result';
+import { Study } from '../../proto/generated/study';
+
+export async function readStudyFile(studyFilePath: string): Promise<Study[]> {
+  const result = await readStudyFileReturnWithError(studyFilePath);
+  if (result.ok) {
+    return result.value[0];
+  } else {
+    throw result.error;
+  }
+}
+
+export async function readStudyFileReturnWithError(
+  studyFilePath: string,
+): Promise<Result<[Study[], string], Error>> {
+  try {
+    const studyArrayString = await fs.readFile(studyFilePath, 'utf8');
+    const studyArray = parseStudyArray(studyArrayString);
+    return Result.ok([studyArray, studyArrayString]);
+  } catch (e) {
+    if (e instanceof Error) {
+      e.message += ` (${studyFilePath})`;
+      return Result.error(e);
+    }
+    // Re-throw non-Error exceptions.
+    throw e;
+  }
+}
+
+export async function writeStudyFile(
+  studyArray: Study[],
+  studyFilePath: string,
+) {
+  await fs.writeFile(studyFilePath, stringifyStudyArray(studyArray));
+}
+
+export function parseStudyArray(studyArrayString: string): Study[] {
+  const jsonStudies = JSON.parse(studyArrayString, jsonStudyReviever);
+  if (!Array.isArray(jsonStudies)) {
+    throw new Error('Root element must be an array');
+  }
+
+  const studyArray = [];
+  for (const jsonStudy of jsonStudies) {
+    const parsedStudy = Study.fromJson(jsonStudy, {
+      ignoreUnknownFields: false,
+    });
+    studyArray.push(parsedStudy);
+  }
+
+  return studyArray;
+}
+
+export function stringifyStudyArray(studyArray: Study[]): string {
+  const jsonStudies: any[] = [];
+  for (const study of studyArray) {
+    const jsonStudy = Study.toJson(study, {
+      emitDefaultValues: false,
+      enumAsInteger: false,
+      useProtoFieldName: true,
+    });
+    jsonStudies.push(jsonStudy);
+  }
+
+  // Use 2 spaces for indentation and add a newline at the end to match Prettier
+  // `json-stringify` behaviour.
+  return JSON.stringify(jsonStudies, jsonStudyReplacer, 2) + '\n';
+}
+
+function jsonStudyReplacer(key: string, value: any): any {
+  switch (key) {
+    case 'start_date':
+    case 'end_date': {
+      const intValue = Number.isInteger(value) ? value : parseInt(value);
+      return new Date(intValue * 1000).toISOString();
+    }
+    case 'channel':
```

Comment:
Could we pass a param isBrave: boolean = true to make the tools work for Finch seed?